rpp30-boc-heartbleed/whimsical
# add user authentication with firebase

## Description
implemented user authentication with firebase

## How to test
Load app in simulator. login with the saved user account info or click register to create a new account.  Fill out the registration form and click 'Create Account.' Once successfully registered with a new account, the user is automatically logged in and will be taken to the Dashboard component. I have added a sign out button for testing purposes in the dashboard. Click the sign out button to sign out. The user will be taken to the login component. Enter the email and password used to register to login in, and the user will be taken to the dashboard where they can click the sign out button to sign out. if an incorrect email or password is entered, an appropriate error message will appear. 

## Notes
Note that /client/.env did not get pushed to github, so please DM me for API keys and user account info.
Note that it takes a moment for firebase to update / initialize (speculating this is the issue), so logging in too fast might cause an error (message will appear to please try again). Try again and user will be logged in and taken to dashboard.
Note that app.json has been changed to app.config.js
Finally, please note that there were some merge conflicts. Conflicts with the Dashboard compononet I fixed manually by accepting the incoming changes and modifying the placement of my signout button / function. Primarily the conflicts were with the package.json and lock files. I deleted the lock files, and the node modules and then reinstalled the node modules. you might find this to be a necessary step as well.

## Issue tracking
(https://trello.com/c/dRsc94VD/36-implement-firebase-authentication)
